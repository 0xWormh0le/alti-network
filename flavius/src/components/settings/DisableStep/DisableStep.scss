@import 'styles/_shared.scss';

.DisableStep {
  &__title {
    color: $TEXT_COLOR;
    margin-bottom: 1rem;
  }

  &__form-group {
    width: 100%;
  }
}
