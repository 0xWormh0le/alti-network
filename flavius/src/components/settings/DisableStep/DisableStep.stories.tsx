import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DisableStep, { DisableStepProps } from './DisableStep'

export default {
  title: 'Settings/DisableStep',
  component: DisableStep
} as Meta

const Template: Story<DisableStepProps> = (args) => <DisableStep {...args} />

export const Default = Template.bind({})
Default.args = {}
