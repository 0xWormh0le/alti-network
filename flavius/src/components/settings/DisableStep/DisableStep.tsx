import React from 'react'
import Button from 'components/elements/Button'
import FormControl, { FormControlProps } from 'components/elements/FormControl'
import FormGroup from 'components/elements/FormGroup'
import FormLabel from 'components/elements/FormLabel'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import UI_STRINGS from 'util/ui-strings'
import { renderAttributeIfDev } from 'util/helpers'
import './DisableStep.scss'

export interface DisableStepProps {
  buttonLoading: boolean
  password: string
  onPasswordChange: (event: React.FormEvent<FormControlProps>) => void
  challengeAnswer: string
  onChallengeAnswerChange: (event: React.FormEvent<FormControlProps>) => void
  onStepCompleted: () => void
}

const DisableStep: React.FC<DisableStepProps> = ({
  buttonLoading,
  password,
  onPasswordChange,
  challengeAnswer,
  onChallengeAnswerChange,
  onStepCompleted
}) => (
  <section className='DisableStep TwoFactorAuthentication__step'>
    <Typography variant={TypographyVariant.H3} weight='semibold' className='DisableStep__title'>
      {UI_STRINGS.SETTINGS.DISABLE_TWO_FACTOR_AUTH}
    </Typography>
    <Typography variant={TypographyVariant.BODY_LARGE}>
      {UI_STRINGS.SETTINGS.ENTER_YOUR_CURRENT_PW_AND_CONFIRM}
    </Typography>
    <Typography variant={TypographyVariant.BODY_LARGE}>
      {UI_STRINGS.SETTINGS.CHANGES_WILL_TAKE_EFFECT_NEXT_TIME}
    </Typography>
    <FormGroup controlId='password' className='DisableStep__form-group'>
      <FormLabel>{UI_STRINGS.SETTINGS.PASSWORD}</FormLabel>
      <FormControl autoFocus={true} type='password' value={password} onChange={onPasswordChange} />
    </FormGroup>
    <FormGroup controlId='challengeAnswer' className='DisableStep__form-group'>
      <FormLabel>{UI_STRINGS.SETTINGS.CONFIRMATION_CODE}</FormLabel>
      <FormControl value={challengeAnswer} onChange={onChallengeAnswerChange} />
    </FormGroup>
    <Button
      className='DisableStep__button'
      disabled={password === '' || challengeAnswer === ''}
      action='primary'
      text={UI_STRINGS.BUTTON_LABELS.CONFIRM}
      loadingText={UI_STRINGS.LOGIN.CONFIRMING}
      isLoading={buttonLoading}
      onClick={onStepCompleted}
      {...renderAttributeIfDev({ 'data-testid': 'confirm-disable-button' })}
    />
  </section>
)

export default DisableStep
