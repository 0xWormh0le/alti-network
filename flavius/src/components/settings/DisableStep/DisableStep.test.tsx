import React from 'react'
import { render } from '@testing-library/react'

import noop from 'lodash/noop'
import DisableStep from './DisableStep'

describe('CongratsStep', () => {
  it('renders correctly', () => {
    const localProps = {
      buttonLoading: false,
      password: 'password',
      onPasswordChange: noop,
      challengeAnswer: 'answer',
      onChallengeAnswerChange: noop,
      onStepCompleted: noop
    }
    const { container } = render(<DisableStep {...localProps} />)
    expect(container).toMatchSnapshot()
  })
})
