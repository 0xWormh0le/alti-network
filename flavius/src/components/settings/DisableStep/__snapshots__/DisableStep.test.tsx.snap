// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CongratsStep renders correctly 1`] = `
<div>
  <section
    class="DisableStep TwoFactorAuthentication__step"
  >
    <h3
      class="Typography Typography--h3 Typography--semibold DisableStep__title"
    >
      Disable Two-Factor Authentication
    </h3>
    <p
      class="Typography Typography--body-large"
    >
      You will need to enter your current password and your confirmation code to disable this feature.
    </p>
    <p
      class="Typography Typography--body-large"
    >
      The changes will take effect the next time you log in. After that, whenever you log in to Altitude Networks app, we will ask you for your username and password only.
    </p>
    <div
      class="FormGroup DisableStep__form-group"
    >
      <label
        class="FormLabel"
        for="password"
      >
        Password
      </label>
      <input
        class="FormControl FormControl--flat"
        id="password"
        type="password"
        value="password"
      />
    </div>
    <div
      class="FormGroup DisableStep__form-group"
    >
      <label
        class="FormLabel"
        for="challengeAnswer"
      >
        Confirmation Code
      </label>
      <input
        class="FormControl FormControl--flat"
        id="challengeAnswer"
        value="answer"
      />
    </div>
    <button
      class="Button DisableStep__button Button--primary"
      data-testid="confirm-disable-button"
      type="button"
    >
      Confirm
    </button>
  </section>
</div>
`;
